claat/parser/parse.go

```diff
@@ -26,12 +26,23 @@ import (
 // Each parser needs to call Register to become a known parser.
 type Parser interface {
 	// Parse parses source r into a Codelab for the specified environment env.
-	Parse(r io.Reader) (*types.Codelab, error)
+	Parse(r io.Reader, opts Options) (*types.Codelab, error)
 
 	// ParseFragment is similar to Parse except it doesn't parse codelab metadata.
 	ParseFragment(r io.Reader) ([]types.Node, error)
 }
 
+// Container for parsing options.
+type Options struct {
+	PassMetadata map[string]bool
+}
+
+func NewOptions() *Options {
```

Comment:
nit: make this take a variadic (read 'optional') arg that carries an non-dummy-initialized `PassMetadata`

Reply:
This becomes progressively harder to manage as the number of fields increases, and it doesn't make a marked difference over users assigning to PassMetadata directly. I think this is better off as an initializer rather than a constructor.

Reply:
sgtm